### PR TITLE
FileErrorHandler

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-719

### Change description ###
Further refactoring BlobProcessorTask - extracted file error handling.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by rkondratowicz** [comment]
Could you provide a description?

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                blobManager.tryMoveFileToRejectedContainer(zipFilename, containerName, leaseId);
+            },
+            () -> {
+                log.error(
+                    "Error notification not sent because Error code mapping not found for {}. "
+                        + "File name: {} Container: {} Reason: {}",
+                    cause.getClass().getName(),
+                    zipFilename,
+                    containerName,
+                    cause
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandler.java:69]: what's going to be logged here?

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            EVENT,
+            CONTAINER,
+            FILE_NAME,
+            MSG,
+            null
+        ))
+            .willReturn(EVENT_ID);
+
+        // when
+        // then
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandlerTest.java:105]: you can use something like
```java
var exc = catchThrowable(() -> fileErrorHandler.handleInvalidFileError(...))
```
and then make assertions on the exception if you want.

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandler.java (hunk 1 above)
Comment (apereverzin) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandler.java:69]: what do you mean @rkondratowicz ? the log message is same as it used to be in BlobProcessorTask

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandlerTest.java (hunk 2 above)
Comment (apereverzin) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/FileErrorHandlerTest.java:105]: ok

**Comment by apereverzin** [comment]
Broken build

**Pull request closed without merging**